src/api/ledger/transaction.js

```diff
@@ -62,11 +62,19 @@ function getTransactionAsync(identifier: string, options: TransactionOptions,
       error = new errors.NotFoundError('Transaction not found');
     }
 
+    const minLedgerVersion = options.minLedgerVersion;
+
     if (error instanceof errors.NotFoundError
-        && !utils.hasCompleteLedgerRange(remote,
-            options.minLedgerVersion, maxLedgerVersion)) {
-      callback(new errors.MissingLedgerHistoryError('Transaction not found,'
-        + ' but the server\'s ledger history is incomplete'));
+        && !utils.hasCompleteLedgerRange(remote, minLedgerVersion,
+                                         maxLedgerVersion)) {
+      if (minLedgerVersion && maxLedgerVersion
+          && utils.isNonContiguousLedgerRange(remote, minLedgerVersion, 
+                                         maxLedgerVersion)) {
+        callback(new errors.NonContiguousLedgerHistoryError());
+      } else if (maxLedgerVersion 
+                 && utils.isPendingLedgerVersion(remote, maxLedgerVersion)) {
+        callback(new errors.PendingLedgerError());
+      }
```

Comment:
> The if/else doesn't cover all the cases.

The cases it covers:

* tx is missing
  * server has incomplete ledger history in the specified range, 
    * ledger history is noncontiguous in the specified range
    * maxLedgerVersion is ahead of the server's last known ledger
  * server has complete ledger history in the specified range
* tx is not missing
  * ...

What's missing?